app/models/host/managed.rb

```diff
@@ -778,6 +779,18 @@ def setup_clone
 
   private
 
+  # validate uniqueness can't prevent saving two interfaces that has same DNS name
+  # because the validation happens before transaction is committed, so data are not in DB
+  # yet, this is the reason why we "reimplement" uniqueness validation
+  def validate_dns_name_uniqueness
+    dups = self.interfaces.group_by { |i| [ i.name, i.domain_id ] }.detect { |_, nics| nics.count > 1 }
```

Comment:
It should be consistent with the `validates :name` in the Nic models themselves, which seems to apply to all Nic::Interface derivatives right now, irrespective of whether they're managed or not, so I think this is correct.  It might be that should change too for unmanaged, but we can probably deal with it separately if it becomes an issue.

I think as long as this validation is consistent with the Nic::* validations, then we'd be unlikely to get into a case where a host becomes invalid, as the Nic validations would prevent conflicting edits?